src/scripts/search-engines/ecosia.ts

```diff
@@ -0,0 +1,49 @@
+import { SearchEngine } from '../types';
+import { createAction, getAddedElements, getEntry, getURL } from './helpers';
+import ecosiaStyle from '!!raw-loader!extract-loader!css-loader!sass-loader!../../styles/search-engines/ecosia.scss';
+
+const pageSelectors = {
+  results: '.mainline-results .result',
+  entry: '.mainline-results .result',
+  url: '.mainline-results .result a',
+  message: '.mainline-top',
+};
+
+const ubSelectors = {
+  action: 'ub-web-general-action',
+  control: 'ub-web-control msg',
+  button: '.ub-link-button',
+};
+
+export const ecosia: SearchEngine = {
+  matches: [ '*://www.ecosia.org/search?*' ],
+  messageNames: {
+    name: 'searchEngines_ecosiaName',
+  },
+  style: ecosiaStyle,
+
+  getHandlers: () => ({
+    controlHandlers: [
+      {
+        createControl: (): HTMLElement | null => {
+          const message = document.querySelector(pageSelectors.message);
+          if (!message) {
+            return null;
+          }
+          const control = document.createElement('div');
+          control.className = ubSelectors.control;
+          message.insertBefore(control, message.firstChild);
+          return control;
+        },
+      },
+    ],
+    entryHandlers: [
+      {
+        getEntry: getEntry(pageSelectors.entry),
+        getURL: getURL(pageSelectors.url),
+        createAction: createAction(ubSelectors.action),
```

Comment:
This line causes a type error, because `createAction` takes two arguments:
`createAction: createAction(ubSelectors.action, ''),`

Reply:
In my environment, lint (`yarn lint`) emits no error.
Please try to fix the above type error. It may be related to the error you got.

Reply:
The unsafe assignment of any value error disappeared without any code changes (weird), but I added the empty string as 2nd param just in case.